### ISSUE TITLE
I can't upload Invoice

### ISSUE THREAD

**Issue by Kernel1453** [jlevers/selling-partner-api#519]
Dear All,
I'm trying to upload invoice but I'm getting error below

```
[400] {
  "errors": [
    {
      "code": "InvalidInput",
      "message": "Attachment with ID 62797d05-a812-4ce1-9315-d596b82e6114 does not exist",
      "details": ""
    }
  ]
}
```

What I'm trying to do is

```php
$request->validate([
                    'file' => 'nullable|file'
                ]);
                $data = null;


                if ($request->hasFile('file')) {
                    $this->addFileDetail($request, $saveMessage, 'file');

                    $attachment = $this->attachments($request, $config, $amazonOrderId, $marketplaceIds, 'file',
                        'invoice');
                    $data['attachments'] = $attachment;
                }

                $body = new InvoiceRequest($data);
                try {
                    $result = $apiInstance->sendInvoice($amazonOrderId, $marketplaceIds, $body);
                    print_r($result);
                } catch (\SellingPartnerApi\ApiException $e) {
                    dd($e->getMessage());

                    $result = 'Exception sendInvoice: ' . $e->getMessage() . PHP_EOL;
                }
```

```php
protected function attachments($request, $config, $amazonOrderId, $marketplaceIds, $fileName, $messageType = null)
    {
        if ($request->hasFile($fileName)) {
            $uploadApi = new UploadsV20201101Api($config);
            $file = base64_encode(file_get_contents($request->file($fileName)->path()));
            $md5Content = md5($file, false);
            $destination = 'messaging/v1/orders/' . $amazonOrderId . '/messages/' . $messageType ?? $request->mesage_type;
            $result = $uploadApi->createUploadDestinationForResource($marketplaceIds, $md5Content, $destination);
            $resultPayload = $result->getPayload();
            $attachment = new Attachment([
                'upload_destination_id' => $resultPayload['upload_destination_id'],
                'file_name' => $request->mesage_type . $amazonOrderId . '.' . $request->file($fileName)->extension()
            ]);

            return [$attachment];
        }
    }
```

```php
protected function addFileDetail($request, $saveMessage, $fileName)
    {
        $saveMessage['file_name'] = $request->file($fileName)->getClientOriginalName();
        $saveMessage['file_extension'] = $request->file($fileName)->getClientOriginalExtension();
    }
```

where i am making mistake, could you please help asap? Thank you.
Best Regards

**Comment by Kernel1453** [comment]
Hi All,
Do you know any solution to overcome this, we are still trying this. @jlevers

**Comment by jlevers** [comment]
@Kernel1453 what is the full response you get when you call `createUploadDestinationForResource`?

**Comment by jlevers** [comment]
Closing due to inactivity. If you still need support on this, let me know and I'll reopen it.

**Comment by Kernel1453** [comment]
Hello, still cant overcome this and need support.

**Comment by jlevers** [comment]
Please answer my previous question so that I can help you:

> @Kernel1453 what is the full response you get when you call `createUploadDestinationForResource`?

**Comment by Kernel1453** [comment]
@jlevers Return message is same.

[400] {
  "errors": [
    {
      "code": "InvalidInput",
      "message": "Attachment with ID XXXXXX does not exist",
      "details": ""
    }
  ]
}

**Comment by jlevers** [comment]
So the error is happening before you get to the `sendInvoice` call? Have you ever reached that call?

**Comment by jlevers** [comment]
What does your `$destination` variable contain if you `var_dump` it right before the `createUploadDestinationForResource` call?

**Comment by Kernel1453** [comment]
@jlevers Hi, thank you for reply my message.
this is my var_dump => string(56) "messaging/v1/orders/406-1310933-XXXXXXX/messages/invoice"

and when i send file returns

{ "message": "Client error: `POST https://sellingpartnerapi-eu.amazon.com/messaging/v1/orders/406-1310933-XXXXXXX/messages/invoice?marketplaceIds=A33AVAJ2PDY3EV` resulted in a `400 Bad Request` response:\n{\n \"errors\": [\n {\n \"code\": \"InvalidInput\",\n \"message\": \"Attachment with ID 083a7d5a-b617-4473-b03c-2d52cb1 (truncated...)\n", "exception": "GuzzleHttp\\Exception\\ClientException", "file":

here error is Attachment with ID doesnt exist.

Could you please check my code below about MD5 hashing and other methods are correct or need to be fixed? again thank you very much.

best regards

**Comment by Kernel1453** [comment]
We still have this issue. Now getting yoursignature doesnt match. What should we do? Please help @jlevers

**Comment by jlevers** [comment]
I'm not positive about this, but I think you need a leading slash in your `$destination` string (so `/messaging/v1/orders/406-1310933-XXXXXXX/messages/invoice` instead of `messaging/v1/orders/406-1310933-XXXXXXX/messages/invoice`).

If that doesn't solve the problem, I recommend contacting Amazon support – I don't use this API so it's hard for me to know what the issue is.

**Comment by Kernel1453** [comment]
hi @jlevers 
when i use /messaging, at this time i get

"message": "[403] {\n  \"errors\": [\n    {\n      \"code\": \"Unauthorized\",\n      \"message\": \"Access to requested resource is denied.\",\n      \"details\": \"\"\n    }\n  ]\n}",
    "exception": "SellingPartnerApi\\ApiException",

**Comment by jlevers** [comment]
Do you have the [Buyer Communication role](https://developer-docs.amazon.com/sp-api/docs/buyer-communication-role)?

**Comment by Kernel1453** [comment]
@jlevers yes, i have all the roles including the buyer communication role.

**Comment by jlevers** [comment]
Weird, ok. I recommend opening a support case with Amazon, because I don't have any further insight into this, and I'm 99% sure it's an API issue, not an issue with this library.

**Comment by Kernel1453** [comment]
@jlevers thank you very much. I will open a case and inform you.

**Comment by iajrz** [comment]
Closing due to inactivity.
